routing/router.go

```diff
@@ -290,6 +292,10 @@ type Config struct {
 	//
 	// TODO(yy): remove it once the root cause of stuck payments is found.
 	ClosedSCIDs map[lnwire.ShortChannelID]struct{}
+
+	// TrafficShaper is an optional traffic shaper that can be used to
+	// control the outgoing channel of a payment.
+	TrafficShaper fn.Option[TlvTrafficShaper]
```

Comment:
Since the traffic shaper is itself an interface, you may have better success giving noop semantics to the nil pointer and then not using an Option here. The ability to do this hangs on whether you can give a satisfactory interpretation of the request if the shaper is nil that doesn't break the promised semantics of the interface.

Reply:
Hmm, I don't think we can enforce that just at the _interface_ level, instead all the implementations would need to handle the nil pointer receive case to return _some_ value. But with the option format, we detect that by just having a `None` option which is then mapped to `None`.